src/Kunstmaan/AdminListBundle/Controller/AdminListController.php

```diff
@@ -322,10 +322,10 @@ protected function doViewAction(AbstractAdminListConfigurator $configurator, $en
             throw $this->createAccessDeniedException('You do not have sufficient rights to access this page.');
         }
 
-        $MetaData = $em->getClassMetadata($configurator->getRepositoryName());
-        $fields = array();
+        $metaData = $em->getClassMetadata($configurator->getRepositoryName());
+        $fields = [];
         $accessor = PropertyAccess::createPropertyAccessor();
-        foreach ($MetaData->fieldNames as $value) {
+        foreach ($metaData->getReflectionProperties() as $value => $reflectionProperty) {
             $fields[$value] = $accessor->getValue($helper, $value);
```

Comment:
You could add a check here to see if the object has a `__toString` method, otherwise skip the property. Something like

```php
if (\is_object($value) && !method_exists($value, '__toString')) {
    continue;
}
```

Reply:
That would not return the expected results. The value could be returned by `$accessor->getValue($helper, $value);` without having `__toString()` implemented in the entity. Adding your suggestion would cause no values to be displayed.

I've just tested this with an entity that has relations without having `__toString()` implemented. It caused an error. But the error came from the method `getStringValue()` from `AbstractAdminListConfigurator`, because it just tries to access `getName()` on every entity in the collection (eventhough that most likely does not exist).

I get that you cant add this if it breaks rendering objects  which don't have the __toString method. I think at this point it should only break for anyone who has overwritten the `getStringValue()` method, while not calling `parent::getStringValue()` and not taking in account the `$result` could be an instance of `PersistentCollection`. I don't know what the policy is that you guys have on this, but if this is not good enough then you can close this pull request :)